src/components/Timer/Timer.js

```diff
@@ -1,18 +1,60 @@
-import React, { useState, useEffect } from 'react';
+import React, { useState, useReducer, useEffect } from 'react';
 import useInterval from './../../hooks/useInterval';
+import convertFromSecondToMinutes from './../../utils/convertFromSecondToMinutes';
+import { ACTIONS } from './../../utils/actions';
+
+const initialState = {
+    pomodoro: 20,
+    shortBreak: 5,
+    longBreak: 10,
+};
+
+const intervalOrder = [
+    'pomodoro',
+    'shortBreak',
+    'pomodoro',
+    'shortBreak',
+    'pomodoro',
+    'shortBreak',
+    'pomodoro',
+    'longBreak',
+];
+
+const reducer = (state, action) => {
+    switch (action.type) {
+        case ACTIONS.POMODORO:
+            return { ...state, pomodoro: state.pomodoro - 1 };
+        case ACTIONS.SHORTBREAK:
+            return { ...state, shortBreak: state.shortBreak - 1 };
+        case ACTIONS.LONGBREAK:
+            return { ...state, longBreak: state.longBreak - 1 };
+        case ACTIONS.RESET:
+            return initialState;
+        default:
+            return state;
+    }
+};
 
 const Timer = () => {
-    const [time, setTime] = useState({
-        pomodoro: 3600,
-        shortBreak: 100,
-        longBreak: 1000,
-    });
-    const [delay] = useState(1000);
+    const [state, dispatch] = useReducer(reducer, initialState);
+    const [delay] = useState(100);
     const [isRunning, setIsRunning] = useState(false);
+    const [intervalElement, setIntervalElement] = useState(0);
+
+    const actualRunningTimer = intervalOrder[intervalElement];
+
+    useEffect(() => {
+        intervalElement === intervalOrder.length && setIntervalElement(0);
+
+        if (state[actualRunningTimer] <= 0) {
+            dispatch({ type: ACTIONS.RESET });
+            setIntervalElement((prev) => prev + 1);
+        }
+    }, [state, actualRunningTimer, intervalElement]);
 
     useInterval(
         () => {
-            setTime({ ...time, pomodoro: time.pomodoro - 1 });
+            dispatch({ type: ACTIONS[actualRunningTimer.toUpperCase()] });
```

Comment:
taking an example, we are converting:

`pomodoro` into `POMODORO` to find a value `pomodoro`

it makes a lot of sense to use the `ACTIONS` here, in fact you should.

but maybe we can make `actualRunningTimer` match the actions without need to `toUpperCase`?